core/trino-spi/src/main/java/io/trino/spi/StandardErrorCode.java

```diff
@@ -185,6 +185,7 @@ public enum StandardErrorCode
     EXCHANGE_MANAGER_NOT_CONFIGURED(65564, INTERNAL_ERROR),
     CATALOG_NOT_AVAILABLE(65565, INTERNAL_ERROR),
     CATALOG_STORE_ERROR(65566, INTERNAL_ERROR),
+    QUERY_EXCEEDED_COMPILER_LIMIT(65567, INTERNAL_ERROR),
```

Comment:
I'm thinking this should be `USER_ERROR`, as it is similar to `QUERY_TEXT_TOO_LARGE` or `QUERY_HAS_TOO_MANY_STAGES`.

It's not necessarily something the user "did wrong" but it's based on their query which is under their control.

Reply:
It seems pretty close to both.  I leaned towards INTERNAL_ERROR but there's a good argument to be made for USER_ERROR.  I'm fine changing it.